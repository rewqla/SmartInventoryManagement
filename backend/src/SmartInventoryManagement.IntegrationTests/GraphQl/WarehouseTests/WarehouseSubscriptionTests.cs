namespace SmartInventoryManagement.IntegrationTests.GraphQl.WarehouseTests;

public class WarehouseSubscriptionTests: IClassFixture<GraphQLServiceSetup>
{
    private readonly GraphQLServiceSetup _graphQlServiceSetup;

    public WarehouseSubscriptionTests(GraphQLServiceSetup graphQlServiceSetup)
    {
        _graphQlServiceSetup = graphQlServiceSetup;
    }

    
    [Fact]
    public async Task WarehouseCreated_SubscriptionReceivesEvent()
    {
        // Arrange & Act
        
        using var cts = new CancellationTokenSource(1_000);
        await using var result = await _graphQlServiceSetup.RequestExecutor.ExecuteAsync(
            @"
                subscription{
                  warehouseCreated{
                    name
                    location
                  }
                }");
        
        await _graphQlServiceSetup.RequestExecutor.ExecuteAsync(
            @"
                mutation{
                  createWarehouse(input:  {
                         name: ""Normal Warehouse"",
                         location: ""Lutskiy""
                      }){
                      name
                      location
                      __typename
                      errors{
                        ...on ValidationError{
                          message
                          errors{
                            propertyName
                            errorMessage
                          }
                        }
                      }
                    }
                  }");
        
        var items = new List<object>();
        var count = 0;

        await foreach(var item in result.ExpectResponseStream()
                          .ReadResultsAsync().WithCancellation(cts.Token))
        {
            items.Add(item.Data);
            count++;
            if (count == 2)
            {
                break;
            }
        }
        
        // Assert
        items.MatchSnapshot();
    }
}