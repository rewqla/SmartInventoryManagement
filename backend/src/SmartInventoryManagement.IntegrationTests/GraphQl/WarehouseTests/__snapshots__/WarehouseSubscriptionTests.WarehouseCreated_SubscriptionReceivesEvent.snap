[
  {
    "warehouseCreated": {
      "name": "Normal Warehouse",
      "location": "Lutskiy"
    }
  }
]
